core/src/erasure.rs

```diff
@@ -470,6 +477,78 @@ pub fn recover(
     Ok((missing_data, missing_coding))
 }
 
+pub fn recover_with_blobs(
+    slot: u64,
+    start_idx: u64,
+    blobs_to_recover: &[Option<Vec<u8>>; NUM_CODING + NUM_DATA],
+) -> Result<(Vec<SharedBlob>, Vec<SharedBlob>)> {
+    let block_start_idx = start_idx - (start_idx % NUM_DATA as u64);
+
+    debug!("block_start_idx: {}", block_start_idx);
+
+    let coding_start_idx = block_start_idx + NUM_DATA as u64 - NUM_CODING as u64;
+    let block_end_idx = block_start_idx + NUM_DATA as u64;
+    trace!(
+        "recover: coding_start_idx: {} block_end_idx: {}",
+        coding_start_idx,
+        block_end_idx
+    );
+
+    let mut blobs: Vec<SharedBlob> = Vec::with_capacity(ERASURE_SET_SIZE);
+    let mut erasures: Vec<i32> = Vec::with_capacity(NUM_CODING);
+
+    let mut missing_data: Vec<SharedBlob> = vec![];
+    let mut missing_coding: Vec<SharedBlob> = vec![];
+
+    if blobs_to_recover[NUM_DATA..].iter().all(Option::is_none) {
```

Comment:
blobs_to_recover is packed, I guess?

Reply:
since below it looks like blobs_to_recover is *not* packed, i.e. that it looks like [DATA; NUM_DATA, CODING, NUM_CODING] this check needs to iterate over the entire slice to verify that there are no more than NUM_CODING blobs missing, be they CODING or DATA.  I think this checks that there is at least one CODING, which is an insufficient check.  the erasure code may do the same check (i.e. verify that erasures.len() < NUM_CODING), but you can save some upfront work.